packages/compiler/test/render3/view/binding_spec.ts

```diff
@@ -98,4 +112,50 @@ describe('t2 binding', () => {
     expect(elDirectives.length).toBe(1);
     expect(elDirectives[0].name).toBe('Dir');
   });
+
+  describe('matching inputs to consuming directives', () => {
+    it('should work for bound attributes', () => {
+      const template = parseTemplate('<div hasInput [inputBinding]="myValue"></div>', '', {});
+      const binder = new R3TargetBinder(makeSelectorMatcher());
+      const res = binder.bind({template: template.nodes});
+      const el = template.nodes[0] as a.Element;
+      const attr = el.inputs[0];
+      const consumer = res.getConsumerOfBinding(attr) as DirectiveMeta;
+      expect(consumer.name).toBe('HasInput');
+    });
+
+    it('should work for text attributes on elements', () => {
+      const template = parseTemplate('<div hasInput inputBinding="{{myValue}}"></div>', '', {});
```

Comment:
By using the `{{}}` interpolation within `inputBinding`, it will actually be transformed into a `BoundAttribute`, not `TextAttribute`. You'd need a truly static text binding to end up with a `TextAttribute`:

```suggestion
      const template = parseTemplate('<div hasInput inputBinding="text"></div>', '', {});
```